fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -578,78 +686,23 @@ private void clearBackendTransactions(TransactionState transactionState) {
         }
     }
 
-    /*
-     * get all txns which is ready to publish
-     * a ready-to-publish txn's partition's visible version should be ONE less than txn's commit version.
-     */
-    public List<TransactionState> getReadyToPublishTransactions() throws UserException {
-        List<TransactionState> readyPublishTransactionState = new ArrayList<>();
-        List<TransactionState> allCommittedTransactionState = null;
+
+    public List<TransactionState> getReadyToPublishTransactions() {
+        List<TransactionState> allCommittedTransactionState;
         writeLock();
```

Comment:
readLock(), and also readUnlock() below.

Reply:
update